cpp/src/db/db_object.cu

```diff
@@ -282,31 +255,28 @@ idx_t db_result<idx_t>::getSize() {
 
 template<typename idx_t>
 idx_t* db_result<idx_t>::getData(std::string idx) {
-  if (!dataValid)
-    throw new std::invalid_argument("Data not valid");
+  CUGRAPH_EXPECTS(dataValid, "Data not valid");
 
   idx_t* returnPtr = nullptr;
   for (size_t i = 0; i < names.size(); i++)
     if (names[i] == idx)
-      returnPtr = columns[i];
+      returnPtr = reinterpret_cast<idx_t*>(columns[i].data());
   return returnPtr;
 }
 
 template<typename idx_t>
 void db_result<idx_t>::addColumn(std::string columnName) {
-  if (dataValid)
-    throw new std::invalid_argument("Cannot add a column to an allocated result");
+  CUGRAPH_EXPECTS(!dataValid, "Cannot add a column to an allocated result.");
   names.push_back(columnName);
 }
 
 template<typename idx_t>
 void db_result<idx_t>::allocateColumns(idx_t size) {
-  if (dataValid)
-    throw new std::invalid_argument("Already allocated columns");
+  CUGRAPH_EXPECTS(!dataValid, "Already allocated columns");
+
   for (size_t i = 0; i < names.size(); i++) {
-    idx_t* colPtr = nullptr;
-    ALLOC_TRY(&colPtr, sizeof(idx_t) * size, nullptr);
-    columns.push_back(colPtr);
+    rmm::device_buffer col(sizeof(idx_t) * size);
+    columns.push_back(std::move(col));
```

Comment:
columns.push_back(rmm::device_buffer(sizeof(idx_t) * size));

This makes it clear that you're passing an rvalue without using std::move. This also prevents users from accidentally using moved-out variable after this statement.